test/libsolidity/SemanticTest.cpp

```diff
@@ -87,9 +87,15 @@ SemanticTest::SemanticTest(string const& _filename, langutil::EVMVersion _evmVer
 	if (m_runWithEwasm && !m_supportsEwasm)
 		m_runWithEwasm = false;
 
-	m_runWithABIEncoderV1Only = m_reader.boolSetting("ABIEncoderV1Only", false);
-	if (m_runWithABIEncoderV1Only && !solidity::test::CommonOptions::get().useABIEncoderV1)
-		m_shouldRun = false;
+	if (!solidity::test::CommonOptions::get().useABIEncoderV1)
+	{
+		for (auto const& [filename,source]: m_reader.sources().sources)
```

Comment:
It would be actually possible to refactor `TestCase` to provide two helpers:
- `sourcesContain(string) -> bool` or `sourcesRequireABIV1() -> bool`
- `sourcesAllowEVMVersion(<version>) -> bool`

This could remove the need for `EVMVersionRestrictedTestCase` too.

However, I'd just go with this approach as in the PR for now.